library/x509_crt.c

```diff
@@ -657,8 +659,11 @@ static int x509_get_subject_alt_name( unsigned char **p,
                     MBEDTLS_ERR_ASN1_UNEXPECTED_TAG );
         }
 
-        /* Skip everything but DNS name */
-        if( tag != ( MBEDTLS_ASN1_CONTEXT_SPECIFIC | 2 ) )
+        tag_mask = tag & ( MBEDTLS_ASN1_TAG_CLASS_MASK | MBEDTLS_ASN1_TAG_VALUE_MASK );
```

Comment:
Minor: Usually a bit mask is a bit pattern that allows you to extract (or insert) bits from a value for parsing or something like that. So in your case the mask is `MBEDTLS_ASN1_TAG_CLASS_MASK | MBEDTLS_ASN1_TAG_VALUE_MASK`. But tag_mask is actually the value extracted from `tag` using that bit mask. I suggest renaming the variable. Also note that the class probably isnt necessary as it was already checked a few lines above.

Reply:
ok